### PR TITLE
[Core] Enforce CA1507 - use nameof() where possible

### PR DESCRIPTION
**Pull request by symbiogenesis**
Enforces [CA1507](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1507)

This will allow potential future bugs to be discovered at compile-time, rather than at runtime.

I used a roslyn refactoring to accomplish this, but I sanity checked all of the individual changes.

### CODE REVIEW

**Review by jsuarezruiz — DISMISSED** [review]

File: src/Compatibility/ControlGallery/src/Issues.Shared/Bugzilla31330.cs [hunk 1]
```diff
@@ -107,7 +107,7 @@ public ListItemViewModel(ListViewModel listViewModel)
 			{
 				if (listViewModel == null)
 				{
-					throw new ArgumentNullException("listViewModel");
+					throw new ArgumentNullException(nameof(listViewModel));
```
Comment (jsuarezruiz) [inline src/Compatibility/ControlGallery/src/Issues.Shared/Bugzilla31330.cs:110]: Can we split the PR in 2-3 ones?, covering:
- Compatibility
- Core
- Controls

Would simplify things to test and review.

**Review by mattleibow — DISMISSED** [review]
I am also thinking that this should actually also only happen in the library code. We have tests that should be fine, but I would rather not change the tests in the same PR that changes the library.

File: src/Compatibility/ControlGallery/src/Issues.Shared/Bugzilla31330.cs (hunk 1 above)
Comment (symbiogenesis) [reply src/Compatibility/ControlGallery/src/Issues.Shared/Bugzilla31330.cs]: Updated this to just target core for now.

**Comment by symbiogenesis** [comment]
> I am also thinking that this should actually also only happen in the library code. We have tests that should be fine, but I would rather not change the tests in the same PR that changes the library.

Removed the changes to the tests, and now this is only impacting Core.

**Review by PureWeen — APPROVED** [review]

**Review by rmarinho — APPROVED** [review]

**Pull request merged**